internal/modules/defaultmodule/defaultmodule.go

```diff
@@ -0,0 +1,92 @@
+package defaultmodule
+
+import (
+	"context"
+	"encoding/json"
```

Comment:
I always wonder whether we should try swapping out for a faster JSON implementation.

Probably this isn't our bottleneck.